### PR TITLE
Fix #1152: Add OfflineAudioContext constructor with dictionary

### LINKED ISSUE #1152: OfflineAudioContext options dictionary?
Now that we have constructors for all of the nodes and they take an options dictionary and even `AudioContext` has an optional dictionary, wouldn't it be nice and consistent if `OfflineAudioContext` also supported a dictionary?

Then you could do
```
new OfflineAudioContext({numberOfChannels: 3, length: 99, sampleRate: 12345})
```
in addition to
```
new OfflineAudioContext(3, 99, 12345)
```

### PR DESCRIPTION
**Pull request by rtoy**
Add constructor for OfflineAudioContext that takes a required
dictionary.  This is an addition to the existing constructor.
    
Define an algorithm that describes how the context is created too.
Update startRendering to match these changes.

### CODE REVIEW

**Comment by rtoy** [comment]
Preview for constructor: https://rawgit.com/WebAudio/web-audio-api/76ab777/index.html#OfflineAudioContext

Preview for dictionary: https://rawgit.com/WebAudio/web-audio-api/76ab777/index.html#offlineaudiocontextoptions

File: index.html [hunk 1]
```diff
@@ -2107,40 +2108,105 @@ <h2 id="OfflineAudioContext">
...
           </dt>
           <dd>
-            Determines the size of the buffer in sample-frames.
+            <p>
+              Let <var>c</var> be a new <a>OfflineAudioContext</a> object.
+              Initialize <var>c</var> as follows:
+            </p>
+            <ol>
+              <li>Set the <code>control thread state</code> for <var>c</var> to
+              <code>suspended</code>.
```
Comment (padenot) [inline index.html]: nit: I think enum values need to be "quoted".

File: index.html [hunk 2]
```diff
@@ -2107,40 +2108,105 @@ <h2 id="OfflineAudioContext">
...
+              </li>
+              <li>Set the <code>rendering thread state</code> for <var>c</var>
+              to <code>suspended</code>.
```
Comment (padenot) [inline index.html]: same

File: index.html [hunk 3]
```diff
@@ -2107,40 +2108,105 @@ <h2 id="OfflineAudioContext">
...
+            Constructor
           </dt>
           <dd>
-            Describes the sample-rate of the linear PCM audio data in the
-            buffer in sample-frames per second. See <a href=
-            "#widl-BaseAudioContext-createBuffer-AudioBuffer-unsigned-long-numberOfChannels-unsigned-long-length-float-sampleRate">
-            createBuffer</a> for valid sample rates.
+            <p>
+              The OfflineAudioContext can constructed with the same arguments
+              as AudioContext.createBuffer. <span class="synchronous">A
```
Comment (padenot) [inline index.html]: nit: Linkify this.

Comment (rtoy) [inline index.html]: Linkify which?  `OfflineAudioContext`?  `AudioContext.createBuffer`?  Both?

**Pull request merged**